app/src/main/java/com/worldtreeinc/leaves/activity/EntryFeeWatcher.java

```diff
@@ -0,0 +1,40 @@
+package com.worldtreeinc.leaves.activity;
+
+import android.text.InputFilter;
+import android.text.Spanned;
+
+/**
+ * Created by chidi on 12/9/15.
+ */
+public class EntryFeeWatcher implements InputFilter {
+    private final int decimaldigits;
+
+    public EntryFeeWatcher(int numberOfDecimalDigits) {
+        this.decimaldigits = numberOfDecimalDigits;
+    }
+
+    @Override
+    public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {
```

Comment:
NITPICK: This block of function could be refactored, what is most important is avoidance of the inner for-loop. Using regex, or looking at other methods for listening such as onfocus would help.